pkg/apiserver/resthandler.go

```diff
@@ -488,8 +488,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		gv := scope.Kind.GroupVersion()
 		codec := runtime.NewCodec(
 			scope.Serializer.EncoderForVersion(s, gv),
-			scope.Serializer.DecoderToVersion(s, unversioned.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
-		)
+			scope.Serializer.DecoderToVersion(s, gv))
```

Comment:
That was how it used to be (for the case when api version obtained from the request scope was v1):

1. Decode v1 -> JSON
2. Apply the patch
3. Decode JSON -> internal

This was wrong because in p.3 JSON was representing external objects, so e.g. if the patch was adding some fields that are only present in ext repr, they were ignored. 

Now it's done like this:

1. Decode v1 -> JSON
2. Apply the patch
3. Decode JSON -> v1
4. Convert v1 -> api

converter and targetGV params were added to support p.4. The object that is passed further is internal.